drivers/ltc4150/include/ltc4150_params.h

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_ltc4150
+ *
+ * @{
+ * @file
+ * @brief       Default configuration for LTC4150 coulomb counters
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef LTC4150_PARAMS_H
+#define LTC4150_PARAMS_H
+
+#include "board.h"
+#include "ltc4150.h"
+#include "saul_reg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name    Set default configuration parameters for the LTC4150
+ * @{
+ */
+#ifndef LTC4150_PARAM_INT
+#define LTC4150_PARAM_INT             (GPIO_PIN(0, 4))
+#endif
+#ifndef LTC4150_PARAM_POL
+#define LTC4150_PARAM_POL             (GPIO_UNDEF)
+#endif
+#ifndef LTC4150_PARAM_SHUTDOWN
+#define LTC4150_PARAM_SHUTDOWN        (GPIO_PIN(0, 5))
+#endif
+#ifndef LTC4150_PARAM_PULSES
+#define LTC4150_PARAM_PULSES          (45700)
```

Comment:
I do not agree on the parentheses remarks. In my opinion there should be a reasonable rule to where to add them. A reasonable rule to could be one of the following to me:

1. add enclosing parentheses in every preprocessor macro, just to be safe
2. add enclosing parentheses  when order of precedence bugs can be introduced

A bug could be introduced e.g. in:

``` c
#define FOO 3 + 4

int bar = 2 * FOO; /* (2 * 3) + 4 not 2 * (3 + 4) */
```

But a precedence bug cannot be introduced by an integer constant. So unless adding enclosing parentheses everywhere, no parentheses are needed here. But if enclosing parentheses would be needed everywhere, they should also be placed around `GPIO_PIN(0, 4)`.

Reply:
fine, but then leave parentheses in place as is.

Reply:
TL;DR: go for rule 1

Reply:
Regarding the `UL` suffix I also disagree. According to the [C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf) in section 6.4.4.1 a decimal constant without any suffix will have the smallest integer type of the following list: `int`, `long int`, `long long int`. Suffixes will only be needed in two cases:

1. The decimal number does fit into `unsigned long long int`, but not into `long long int`
2. The decimal number is used in expressions in which could overflow the automatically determent type of the number, e.g. `uint64_t foo = 42 << 32;`

(Explaining 2.: The 42 would fit into `int`, thus this would be the determent type. However, shifting any `int` by 32 bits to the left will result in `0`. Thus, `foo` will have the value `0` here. Using `uint64_t foo = 42ULL << 32` would have the expected result - so here the suffix would be needed for correct operation.)

Suffixes are really rarely needed. But they make reading the code harder. Adding them only were  actually harm could occur will be more readable in my opinion and raise awareness of the actual reason, why the suffixes are needed in some places.

Reply:
to clarify (from my side) on the suggested `UL` suffix: 

1. we had lots of issue with signed/unsigned comparison errors/warnings, hence the `U`
2. RIOT also supports 8bit platforms such as the atmega arduinos and (IIRC) it was mostly required to explicitly state `UL`, bc w/o any suffix or just `U` `45700 ` will not fit into a 16bit int.

Reply:
Good point with the signed/unsigned comparison. There should be no issue with the Arduino, as the compiler will (according to the standard) just fall back to `long int` or even `long long int` if the decimal number is too big.

But honestly: I believe if the compiler does complain about signed/unsigned comparison with a constant decimal, I would consider this as a compiler bug. The compiler is aware of the value of the constant decimal number and of the type of the other side of the comparison. This should be enough to tell for sure if the comparison will work as expected or not.